### PR TITLE
Fix content type determination for blob from fetch

### PR DESCRIPTION
**Pull request by jasnell**
Using a compatibility flag, selects the correct content type for a blob from the fetch-provided content-type header(s) as defined by the fetch spec: https://fetch.spec.whatwg.org/#concept-header-extract-mime-type

Sadly it's potentially breaking change so the compat flag is needed.

Fixes: https://github.com/cloudflare/workerd/issues/2099

Goal: Web standards compliance, bug fix

### CODE REVIEW

File: src/workerd/util/mimetype.c++ [hunk 1]
```diff
@@ -378,4 +383,64 @@ bool MimeType::isVideo(const MimeType& mimeType) {
...
+  kj::Maybe<MimeType> mimeType;
+
+  constexpr static auto findNextSeparator = [](auto& input) -> kj::Maybe<size_t> {
+    // Scans input to find the next comma (,) that is not contained within
+    // a quoted section, returning the position of the comma or kj::none
+    // if not found.
+    for (size_t i = 0; i < input.size(); ++i) {
+      if (input[i] == '"') {
+        // Skip to the end of the quoted section
+        while (++i < input.size() && (input[i] != '"' || input[i-1] == '\\')) {}
```
Comment (dom96) [inline src/workerd/util/mimetype.c++:397]: I'm confused by the handling of inputs like `\"foobar` here. Doesn't that indicate that it's not a quoted section in which case nothing should be skipped? But this code as it is right now will skip to the end of the "foobar", am I misreading something or is this incorrect?

**Review by dom96 — APPROVED** [review]
Nice! Looks good overall but have one question.

File: src/workerd/util/mimetype.c++ (hunk 1 above)
Comment (jasnell) [reply src/workerd/util/mimetype.c++:397]: ah, likely need an additional condition on the previous line to ensure the `"` we're on currently isn't escaped. Will update and add a test (that edge case is apparently not covered by the existing web platform tests)
